docker2earthly/convert.go

```diff
@@ -139,3 +142,71 @@ func Docker2Earthly(dockerfilePath, earthfilePath, imageTag string) error {
 	fmt.Fprintf(out, "\nbuild:\n    BUILD +subbuild%d\n", i)
 	return nil
 }
+
+var earthfileTemplate = `
+VERSION --use-docker-ignore {{.Version}}
+# This Earthfile was generated using {{.CommandName}} command
+docker:
+	{{- range .BuildArgs}}
+	ARG {{. -}}
+	{{- end}}
+	FROM DOCKERFILE \
+	{{- range .BuildArgs}}
+	--build-arg {{.}}=${{.}} \
+	{{- end}}
+   	{{- if .Target}}
+	--target {{.Target}} \
+	{{- end}}
+	-f {{.Dockerfile}} \
+	{{.BuildContext}}
+	{{- if .ImageTags}}
+	SAVE IMAGE --push
+	{{- range .ImageTags}} {{. -}}{{- end}}
+	{{- end}}
+
+build:
+	BUILD{{- range .Platforms}} --platform {{ . -}}{{- end}} +docker
+`
+
+type earthfileTemplateArgs struct {
+	Version      string
+	CommandName  string
+	BuildArgs    []string
+	Target       string
+	Dockerfile   string
+	BuildContext string
+	ImageTags    []string
+	Platforms    []string
+}
+
+// GenerateEarthfileContent returns an Earthfile content string which contains a target to build a docker image using FROM DOCKERFILE
+func GenerateEarthfileContent(buildContextPath string, dockerfilePath string, imageTags []string, buildArgs []string, platforms []string, target string) (string, error) {
```

Comment:
nit: `docker2earthly.GenerateEarthfileContent` is pretty verbose to my eyes. We're already in the `docker2earthly` package, so maybe just `Generate` gives us most of the same information in 1/3 of the characters?

Reply:
I went with `GenerateEarthfile`